controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -826,8 +880,40 @@ func (s regServiceSubstatusHandler) addRegistrationServiceHealthStatus(reqLogger
 		return false
 	}
 
+	// we can set component ready condition to true
 	componentReadyCondition := status.NewComponentReadyCondition(toolchainv1alpha1.ToolchainStatusRegServiceReadyReason)
 	toolchainStatus.Status.RegistrationService.Health.Conditions = []toolchainv1alpha1.Condition{*componentReadyCondition}
+
+	// if we are running in production we also
+	// check that deployed version matches source code repository commit
+	toolchainConfig, errToolchainConfig := toolchainconfig.GetToolchainConfig(s.controllerClient)
+	if errToolchainConfig != nil {
+		reqLogger.Error(errToolchainConfig, status.ErrMsgCannotGetDeployment)
+		errCondition := status.NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusDeploymentRevisionCheckOperatorErrorReason,
+			fmt.Sprintf("unable to get ToolchainConfig: %s", errToolchainConfig.Error()))
+		toolchainStatus.Status.RegistrationService.RevisionCheck.Conditions = []toolchainv1alpha1.Condition{*errCondition}
+		return false
+	}
+
+	// validate deployed version
+	isProd := isProdEnvironment(toolchainConfig)
+	githubRepo := client.GitHubRepository{
+		Org:               toolchainv1alpha1.ProviderLabelValue,
+		Name:              registrationServiceRepoName,
+		Branch:            registrationServiceRepoBranchName,
+		DeployedCommitSHA: healthValues.Revision,
```

Comment:
as for your original mistake of using the host-operator revision - is there an easy way of covering it by unit-test?